modules/alookup/alookup.go

```diff
@@ -40,55 +41,71 @@ func (s *Lookup) DoLookup(name string) (interface{}, zdns.Status, error) {
 	return s.DoTargetedLookup(name, nameServer)
 }
 
-func (s *Lookup) DoTargetedLookup(name, nameServer string) (interface{}, zdns.Status, error) {
-	requests := []uint16{}
-	if s.Factory.Factory.IPv6Lookup {
-		requests = append(requests, dns.TypeAAAA)
+func (s *Lookup) doLookupProtocol(name string, nameServer string, dnsType uint16, searchSet *map[string][]miekg.Answer, origName string, depth int) ([]string, zdns.Status, error) {
```

Comment:
Don't pass a map by pointer, just pass it by value. Maps are reference values. See http://stackoverflow.com/questions/14928826/passing-pointers-to-maps-in-golang